api/anilist/src/main/kotlin/com/imashnake/animite/api/anilist/AnilistApiModule.kt

```diff
@@ -1,40 +1,36 @@
-package com.imashnake.animite.di
+package com.imashnake.animite.api.anilist
 
 import android.content.Context
 import com.apollographql.apollo3.ApolloClient
 import com.apollographql.apollo3.cache.normalized.api.MemoryCacheFactory
 import com.apollographql.apollo3.cache.normalized.normalizedCache
 import com.apollographql.apollo3.cache.normalized.sql.SqlNormalizedCacheFactory
 import com.apollographql.apollo3.network.http.LoggingInterceptor
-import com.imashnake.animite.dev.internal.Constants
 import dagger.Module
 import dagger.Provides
 import dagger.hilt.InstallIn
 import dagger.hilt.android.qualifiers.ApplicationContext
 import dagger.hilt.components.SingletonComponent
-import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.Dispatchers
 import javax.inject.Singleton
 
 @Module
 @InstallIn(SingletonComponent::class)
-object NetworkModule {
-    @Provides
-    fun providesNetworkDispatcher(): CoroutineDispatcher = Dispatchers.IO
+object AnilistApiModule {
 
+    // TODO name this so we can have other apollo clients for different APIs
     @Provides
     @Singleton
     fun provideApolloClient(
-        @ApplicationContext context: Context,
-        dispatcher: CoroutineDispatcher
+        @ApplicationContext context: Context
     ): ApolloClient {
         // Cache is hit in order, so check in-memory -> check sqlite
         // We have an in-memory cache first for speed, then a SQLite cache for persistence.
         val cacheFactory = MemoryCacheFactory(maxSizeBytes = 10 * 1024 * 1024)
             .chain(SqlNormalizedCacheFactory(context, "apollo.db"))
         return ApolloClient.Builder()
-            .dispatcher(dispatcher)
-            .serverUrl(Constants.ANILIST_BASE_URL)
+            .dispatcher(Dispatchers.IO)
+            .serverUrl("https://graphql.anilist.co/")
```

Comment:
if the only thing that would change between this ApolloClient and another being the serverUrl, could easily do named provider arguments, but I don't even know if I wanna use Apollo for regular requests, I use Ktor for the auth-related calls tbh

Reply:
I imagine that would be abstracted away in this module, there's be an `AuthClient` of some kind. They only need to be named because Hilt doesn't allow for internal visibility like that